pkg/detectors/salesmate/salesmate.go

```diff
@@ -0,0 +1,85 @@
+package salesmate
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	//Make sure that your group is surrounded in boundry characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"salesmate"}) + `\b([0-9Aa-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\b`)
+	domainPat = regexp.MustCompile(detectors.PrefixRegex([]string{"salesmateDomain"}) + `\b([a-z0-9A-Z._]{4,22})\b`)
```

Comment:
suggestion: `salesmateDomain `is unlikely to be found, recommended to just use `salesmate`. After some testing, they allow a minimum of 3 characters and not working with special characters like `_ or .` Maybe we can go with `\b([a-z0-9A-Z]{3,22})\b`?